erpnext/hr/doctype/leave_type/leave_type.json

```diff
@@ -211,15 +213,30 @@
    "fieldtype": "Float",
    "label": "Fraction of Daily Salary per Leave",
    "mandatory_depends_on": "eval:doc.is_ppl == 1"
+  },
+  {
+   "default": "0",
+   "depends_on": "eval: doc.include_holiday",
+   "fieldname": "exclude_non_weekly_off_holidays_within_leaves",
+   "fieldtype": "Check",
+   "label": "Exclude non weekly off holidays within leaves"
+  },
+  {
+   "default": "0",
+   "depends_on": "eval: doc.include_holiday",
+   "fieldname": "exclude_public_holidays_within_leaves",
+   "fieldtype": "Check",
+   "label": "Exclude public holidays within leaves"
   }
```

Comment:
It will not make Include holidays within leaves as leaves useless. I have explain with an example

Weekly Offs are checked as weekly off in the holidays table. Lets take an example of a week where:

1. Sat and Sun are weekly off
2. Tuesday to Thursday are public holidays
3. Company decided to give Monday as a holiday as well

Include holidays within leaves as leaves checkbox is important to implement sandwich policy
Exclude non weekly off holidays within leaves is important in the case above where company would want to not deduct as un paid leaves and similarly for the public leaves.

I can give you example of KSA. It is law there that you can not deduct public holidays even it is a sandwich leave policy. So we need to exclude those holidays